usr/share/rear/prep/default/320_include_uefi_env.sh

```diff
@@ -47,13 +47,21 @@ if grep -qw efivars /proc/mounts; then
     SYSFS_DIR_EFI_VARS=/sys/firmware/efi/efivars
 fi
 
-# next step, is checking /boot/efi directory (we need it)
-if [[ ! -d /boot/efi ]]; then
-    return    # must be mounted
+# next step, is case-sensitive checking /boot for case-insensitive /efi directory (we need it)
+if [[ -n `find /boot -maxdepth 1 -iname efi -type d` ]]; then
+    return    # not found
 fi
 
 # next step, check filesystem partition type (vfat?)
+local efi_mount_point=""
 UEFI_FS_TYPE=$(awk '/\/boot\/efi/ { print $3 }' /proc/mounts)
+# if not mounted at /boot/efi, try /boot itself
+if [[ -z "$UEFI_FS_TYPE" ]]; then
+    UEFI_FS_TYPE=$(awk '/\/boot/ { print $3 }' /proc/mounts)
+    [[ -z "$UEFI_FS_TYHPE" ]] && efi_mount_point='/\/boot/'
```

Comment:
Two things.

1. $UEFI_FS_TY**H**PE  I guess this is typo?

2. Even if it is _$UEFI_FS_TYPE_, this part does not give any sense to me. As (even without knowing code context in more detail) following code
```
 # ESP must be type vfat (under Linux)
  if [[ "$UEFI_FS_TYPE" != "vfat" ]]; then
     return
 fi
```

will evaluate as TRUE (as empty != "vfat"), hence ReaR will consider legacy boot is in use so _$efi_mount_point_ becomes useless.

V.